src/main/java/org/springframework/data/web/ProxyingHandlerMethodArgumentResolver.java

```diff
@@ -128,4 +143,33 @@ protected Object createAttribute(String attributeName, MethodParameter parameter
 
 	@Override
 	protected void bindRequestParameters(WebDataBinder binder, NativeWebRequest request) {}
+
+	/**
+	 * Logs a warning message when a parameter is proxied but not explicitly annotated with {@link @ProjectedPayload}.
+	 * <p>
+	 * To avoid log spamming, the message is only logged the first time the parameter is encountered.
+	 */
+	static class ProjectedPayloadDeprecationLogger {
+
+		private static final String MESSAGE = "Parameter %s (%s) is not annotated with @ProjectedPayload - support for parameters not explicitly annotated with @ProjectedPayload (at the parameter or type level) will be dropped in a future version.";
+
+		private final LogAccessor logger = new LogAccessor(ProxyingHandlerMethodArgumentResolver.class);
```

Comment:
I :heart `LogAccessor` as it provides a `Supplier` variant of the log methods which avoids the `if (log.isXYZEnabled...`.